Content.Client/GameObjects/Components/Weapons/ClientFlashableComponent.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.Threading;
+using Content.Shared.GameObjects.Components.Weapons;
+using Robust.Client.Graphics.Drawing;
+using Robust.Client.Graphics.Overlays;
+using Robust.Client.Interfaces.Graphics;
+using Robust.Client.Interfaces.Graphics.Overlays;
+using Robust.Client.Player;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.Timing;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Utility;
+using Timer = Robust.Shared.Timers.Timer;
+
+namespace Content.Client.GameObjects.Components.Weapons
+{
+    [RegisterComponent]
+    public sealed class ClientFlashableComponent : SharedFlashableComponent
+    {
+        private CancellationTokenSource _cancelToken;
+        private TimeSpan _startTime;
+        private double _duration;
+        private FlashOverlay _overlay;
+
+        public override void HandleComponentState(ComponentState curState, ComponentState nextState)
+        {
+            if (curState == null)
+            {
+                return;
+            }
+
+            var playerManager = IoCManager.Resolve<IPlayerManager>();
+            if (playerManager.LocalPlayer.ControlledEntity != Owner)
+            {
+                return;
+            }
```

Comment:
It looks like this check doesn't do anything?

Reply:
It's possible for the client to get states for other entities. If there's a way server-side to only send the state to one player that'd probably be better.